Description: Non-Linux architectures on Debian support to be considered as Linux in the Colobot case
Author: Didier Raboud <odyx@debian.org>
Last-Update: 2013-11-26
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,8 +48,8 @@
 
     # Platform-dependent implementation of system.h
     set(SYSTEM_CPP_MODULE "system_windows.cpp")
-elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
-    message(STATUS "Build for Linux system")
+elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Linux" OR "${CMAKE_SYSTEM_NAME}" MATCHES "kFreeBSD" OR "${CMAKE_SYSTEM_NAME}" STREQUAL "GNU")
+    message(STATUS "Build for GNU/Linux, GNU/kFreeBSD or GNU/Hurd system")
     set(PLATFORM_WINDOWS 0)
     set(PLATFORM_LINUX   1)
     set(PLATFORM_MACOSX  0)
